ambry-router/src/test/java/com.github.ambry.router/GetBlobOperationTest.java

```diff
@@ -0,0 +1,596 @@
+
+  public GetBlobOperationTest() {
+    // Defaults. Tests should override these as appropriate.
+    maxChunkSize = random.nextInt(1024 * 1024) + 1;
+    blobSize = maxChunkSize * random.nextInt(10) + random.nextInt(maxChunkSize - 1) + 1;
+  }
+
+  @After
+  public void after() {
+    if (networkClient != null) {
+      networkClient.close();
+    }
+    Assert.assertEquals("All operations should have completed", 0, operationsCount.get());
+  }
+
+  /**
+   * Instantiate a router, perform a put, close the router. The blob that was put will be saved in the MockServer,
+   * and can be queried by the getBlob operations in the test.
+   */
+  private void doPut()
+      throws Exception {
+    requestParallelism = 3;
+    successTarget = 2;
+    mockSelectorState.set(MockSelectorState.Good);
+    VerifiableProperties vprops = new VerifiableProperties(getNonBlockingRouterProperties());
+    routerConfig = new RouterConfig(vprops);
+    mockClusterMap = new MockClusterMap();
+    routerMetrics = new NonBlockingRouterMetrics(mockClusterMap.getMetricRegistry());
+    mockServerLayout = new MockServerLayout(mockClusterMap);
+    replicasCount = mockClusterMap.getWritablePartitionIds().get(0).getReplicaIds().size();
+    responseHandler = new ResponseHandler(mockClusterMap);
+    networkClientFactory = new MockNetworkClientFactory(vprops, mockSelectorState, MAX_PORTS_PLAIN_TEXT, MAX_PORTS_SSL,
+        CHECKOUT_TIMEOUT_MS, mockServerLayout, time);
+    router =
+        new NonBlockingRouter(routerConfig, new NonBlockingRouterMetrics(new MetricRegistry()), networkClientFactory,
+            new LoggingNotificationSystem(), mockClusterMap, time);
+    blobProperties = new BlobProperties(blobSize, "serviceId", "memberId", "contentType", false, Utils.Infinite_Time);
+    userMetadata = new byte[BLOB_USER_METADATA_SIZE];
+    random.nextBytes(userMetadata);
+    putContent = new byte[blobSize];
+    random.nextBytes(putContent);
+    ReadableStreamChannel putChannel = new ByteBufferReadableStreamChannel(ByteBuffer.wrap(putContent));
+    blobIdStr = router.putBlob(blobProperties, userMetadata, putChannel).get();
+    router.close();
+    networkClient = networkClientFactory.getNetworkClient();
+    resetConfigsForGet();
+  }
+
+  /**
+   * Recreate the configs for the router to be used by gets.
+   */
+  private void resetConfigsForGet() {
+    requestParallelism = 2;
+    successTarget = 1;
+    VerifiableProperties vprops = new VerifiableProperties(getNonBlockingRouterProperties());
+    routerConfig = new RouterConfig(vprops);
+  }
+
+  /**
+   * Test {@link GetBlobOperation} instantiation and validate the get methods.
+   * @throws Exception
+   */
+  @Test
+  public void testInstantiation()
+      throws Exception {
+    maxChunkSize = random.nextInt(1024 * 1024) + 1;
+    blobSize = maxChunkSize * 5;
+    doPut();
+    String blobIdStr = (new BlobId(mockClusterMap.getWritablePartitionIds().get(0))).getID();
```

Comment:
minor: there is already a `blobIdStr` that was set by the `doPut()`. Why the new one?